app/controllers/reports/subscription_reports_for_project_owners_controller.rb

```diff
@@ -0,0 +1,24 @@
+require 'csv'
+
+class Reports::SubscriptionReportsForProjectOwnersController < Reports::BaseController
+  def index
+    if policy(Project.find(params[:project_id])).update?
```

Comment:
I don't like to handle flows using exceptions, imho it adds unnecessary complexity to the code. BUT I agree that the `redirect` here isn't semantic, if the person doesn't have access to the resource, you could simple return `head 401`.